backends/magma/magma_weight.c

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+#include "ceed-magma.h"
+
+//////////////////////////////////////////////////////////////////////////////////////////
+#ifdef __cplusplus
+CEED_INTERN "C"
+#endif
+magma_int_t
+magma_weight(
+  magma_int_t Q, magma_int_t dim,
+  const CeedScalar *dqweight1d,
+  CeedScalar *dV, magma_int_t v_stride,
+  magma_int_t nelem, magma_kernel_mode_t kernel_mode, magma_int_t *maxthreads,
+  magma_queue_t queue) {
+  magma_int_t launch_failed = 0;
+
+  if(kernel_mode == MAGMA_KERNEL_DIM_SPECIFIC) {
+    switch(dim) {
+    case 1: launch_failed = magma_weight_1d(Q, dqweight1d, dV, v_stride, nelem,
+                                              maxthreads[0], queue); break;
+    case 2: launch_failed = magma_weight_2d(Q, dqweight1d, dV, v_stride, nelem,
+                                              maxthreads[1], queue); break;
+    case 3: launch_failed = magma_weight_3d(Q, dqweight1d, dV, v_stride, nelem,
+                                              maxthreads[2], queue); break;
+    default: launch_failed = 1;
```

Comment:
same

Reply:
Please check ef559505b1cd7f4a11060d1f711aa5972054caad. If it looks better, I will propagate this style to the other files

Reply:
We usually have a space like `switch (Q) {`, but the style looks fine to me.